### PR TITLE
Add curation for the Trait protocol.

### PR DESCRIPTION
**Pull request by medreisbach**
Add curation for the Trait protocol.

Part 4 of changes for rdar://119702877

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Review by chuckdude — APPROVED** [review]
This looks great! Thanks, Matt.

File: Sources/Testing/Testing.docc/Traits/Trait.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+- ``Trait/tags(_:)-505n9``
+- ``Trait/tags(_:)-yg0i``
+
+### Associating issues
+
+- ``Trait/bug(_:relationship:_:)-86mmm``
+- ``Trait/bug(_:relationship:_:)-3hsi5``
+
+### Adding comments
+- ``Trait/comment(_:)``
```
Comment (grynspan) [inline Sources/Testing/Testing.docc/Traits/Trait.md:28]: Should we move the `comments` property here?

**Review by grynspan — APPROVED** [review]
Looks good, just one question (not a blocker for merging.)

File: Sources/Testing/Testing.docc/Traits/Trait.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+### Limiting the running time of tests
+
+- ``Trait/timeLimit(_:)``
+ 
+### Categorizing tests
+
+- ``Trait/tags(_:)-505n9``
+- ``Trait/tags(_:)-yg0i``
+
+### Associating issues
```
Comment (grynspan) [inline Sources/Testing/Testing.docc/Traits/Trait.md:22]: Use "bugs" here. An "issue" in swift-testing's DSL is an instance of `Issue`.

**Review by grynspan — CHANGES_REQUESTED** [review]
Sorry, just noticed one thing that needs correcting. :(

File: Sources/Testing/Testing.docc/Traits/Trait.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+- ``Trait/tags(_:)-505n9``
+- ``Trait/tags(_:)-yg0i``
+
+### Associating bugs
+
+- ``Trait/bug(_:relationship:_:)-86mmm``
+- ``Trait/bug(_:relationship:_:)-3hsi5``
+
+### Adding comments
+- ``Trait/comments``
```
Comment (grynspan) [inline Sources/Testing/Testing.docc/Traits/Trait.md:28]: I meant the property _and_ the factory function, sorry!

Comment (grynspan) [reply Sources/Testing/Testing.docc/Traits/Trait.md:28]: Actually let's just switch it back to the function—I think we need to think about the corresponding properties for _all_ of these traits. Sorry for the noise.

Comment (grynspan) [reply Sources/Testing/Testing.docc/Traits/Trait.md:28]: Yeah ignore me here. I think I confused myself.

**Review by grynspan — APPROVED** [review]

**Pull request merged**